### PR TITLE
[SDP-1115] dev: fix forgot password link

### PR DESCRIPTION
**Pull request by CaioTeixeira95**
### What

When the URL doesn't have `protocol` but has the host port number the `url.JoinPath` function doesn't join the URIs.

https://go.dev/play/p/FGhCOuhnObJ
```go
func main() {
	baseURL := "bluecorp.stellar.local:3000"
	u, err := url.JoinPath(baseURL, "forgot-password")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(u) // bluecorp.stellar.local:3000
}
```

https://go.dev/play/p/nvb1-FZWmp4
```go
func main() {
	baseURL := "bluecorp.stellar.local"
	u, err := url.JoinPath(baseURL, "forgot-password")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(u) // bluecorp.stellar.local/forgot-password
}
```

https://go.dev/play/p/thVyyomor5W
```go
func main() {
	baseURL := "http://bluecorp.stellar.local:3000"
	u, err := url.JoinPath(baseURL, "forgot-password")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(u) // http://bluecorp.stellar.local:3000/forgot-password
}
```

python3.10.2 example
```py
>>> from urllib.parse import urljoin
>>> urljoin("bluecorp.stellar.local:3000", "forgot-password")
'forgot-password'
>>> urljoin("bluecorp.stellar.local", "forgot-password")
'forgot-password'
>>> urljoin("http://bluecorp.stellar.local:3000", "forgot-password")
'http://bluecorp.stellar.local:3000/forgot-password'
```

### Why

The forgot-password link was broken.

### Known limitations

N/A

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: dev/main.sh [hunk 1]
```diff
@@ -47,8 +47,8 @@ if [ "$existingTenants" == "[]" ]; then
     for tenant in "${tenants[@]}"
     do
         echo "🐈Provisioning tenant: $tenant"
-        baseURL="$tenant.stellar.local:8000"
-        sdpUIBaseURL="$tenant.stellar.local:3000"
+        baseURL="http://$tenant.stellar.local:8000"
+        sdpUIBaseURL="http://$tenant.stellar.local:3000"
```
Comment (marcelosalloum) [inline dev/main.sh:51]: Since there are other ways to input this information (through APIs and CLI), we should actually change the code so our join logic can work for URLs that don't have the schema

**Review by marcelosalloum — COMMENTED** [review]
I believe the best fix here is to make the code resilient to URLs without the schema.

File: dev/main.sh (hunk 1 above)
Comment (CaioTeixeira95) [reply dev/main.sh:51]: In my opinion, we should reject `BaserURL` and `SDPUIBaseURL` without the scheme/protocol because there's no way to determine which scheme we should use (`HTTP` or `HTTPS`). Relying on the standard library to join the URI is the best since it's battle-tested.

Comment (marcelosalloum) [reply dev/main.sh:51]: I defer this to @marwen-abid

Comment (marcelosalloum) [inline dev/main.sh:51]: Do we actually have https in place for this? I think it'd be `http`, right?

File: internal/crashtracker/sentry_client_test.go [hunk 2]
```diff
@@ -41,25 +41,6 @@ func (m *mockHubSentry) Recover(err interface{}) *sentry.EventID {
...
-	return m.Called(ctx).Get(0).(*mockHubSentry)
-}
-
-func (m *mockSentry) CurrentHub() hubSentryInterface {
-	return m.Called().Get(0).(*mockHubSentry)
-}
-
-// Ensuring that *mockSentry is implementing sentryInterface interface.
-var _ sentryInterface = (*mockSentry)(nil)
-
```
Comment (marcelosalloum) [inline internal/crashtracker/sentry_client_test.go:62]: Thanks for removing that!

File: stellar-multitenant/pkg/internal/validators/tenant_validator.go [hunk 3]
```diff
@@ -104,12 +104,22 @@ func (tv *TenantValidator) ValidateUpdateTenantRequest(reqBody *UpdateTenantRequ
 		if _, err = url.ParseRequestURI(*reqBody.BaseURL); err != nil {
 			tv.Check(false, "base_url", "invalid base URL value")
 		}
+
+		*reqBody.BaseURL, err = utils.SetURLSchemeIfEmpty(*reqBody.BaseURL)
+		if err != nil {
+			tv.Check(false, "base_url", "invalid base URL value. Verify the URL protocol scheme")
```
Comment (marcelosalloum) [inline stellar-multitenant/pkg/internal/validators/tenant_validator.go:110]: You're getting rid of the error here. Can you at least log the error since it's not being added to the check message?

File: stellar-multitenant/pkg/internal/validators/tenant_validator_test.go [hunk 4]
```diff
@@ -195,8 +195,8 @@ func TestTenantValidator_ValidateUpdateTenantRequest(t *testing.T) {
 		tv := NewTenantValidator()
 		invalidValue := "invalid"
 		reqBody := &UpdateTenantRequest{
-			BaseURL:      &invalidValue,
-			SDPUIBaseURL: &invalidValue,
+			BaseURL:      &[]string{invalidValue}[0],
+			SDPUIBaseURL: &[]string{invalidValue}[0],
```
Comment (marcelosalloum) [inline stellar-multitenant/pkg/internal/validators/tenant_validator_test.go:199]: Is this change necessary?

File: internal/utils/url.go [hunk 5]
```diff
@@ -72,3 +73,19 @@ func VerifySignedURL(signedURL string, expectedPublicKey string) (bool, error) {
 
 	return true, nil
 }
+
+func SetURLSchemeIfEmpty(rawURL string) (string, error) {
```
Comment (marcelosalloum) [inline internal/utils/url.go:77]: Calling it a setter is inaccurate because a setter should be modifying an existing resource, while this function returns a new resource.

What about calling it `GetURLWithScheme` or `GetURLStringWithScheme`?

File: stellar-multitenant/pkg/internal/validators/tenant_validator.go (hunk 3 above)
Comment (marcelosalloum) [inline stellar-multitenant/pkg/internal/validators/tenant_validator.go:110]: This extra check would override the previous one, right? I believe you should only apply the second `"base_url"` check if the first one has not returned an error.

File: stellar-multitenant/pkg/internal/validators/tenant_validator.go [hunk 6]
```diff
@@ -104,12 +104,22 @@ func (tv *TenantValidator) ValidateUpdateTenantRequest(reqBody *UpdateTenantRequ
...
 
 	if reqBody.SDPUIBaseURL != nil {
 		if _, err = url.ParseRequestURI(*reqBody.SDPUIBaseURL); err != nil {
 			tv.Check(false, "sdp_ui_base_url", "invalid SDP UI base URL value")
 		}
+
+		*reqBody.SDPUIBaseURL, err = utils.SetURLSchemeIfEmpty(*reqBody.SDPUIBaseURL)
+		if err != nil {
+			tv.Check(false, "sdp_ui_base_url", "invalid SDP UI base URL value. Verify the URL protocol scheme")
+		}
```
Comment (marcelosalloum) [inline stellar-multitenant/pkg/internal/validators/tenant_validator.go:122]: Same here.

**Review by marcelosalloum — APPROVED** [review]
LGTM! Plz, feel free to merge it after addressing the comments below.

File: dev/main.sh (hunk 1 above)
Comment (CaioTeixeira95) [reply dev/main.sh:51]: Sorry, I don't get it.
